### PR TITLE
Make it easier to subclass RecordSensor.py

### PR DESCRIPTION
**Pull request by marionleborgne**
This PR supplements an addition to nupic.research: I needed to subclass RecordSensor.py to create a new sensor region that has the flexibility to be fed data manually [1] so I slightly refactored `RecordSensor.py`. Basically I just took out a chunk of code (that could have used its own method anyway) into a new method that I named `self.applyFilters`. I'm calling this method in the subclass when I override `self.getNextRecord`.

[1] Use case described in the companion `nupic.research` PR https://github.com/numenta/nupic.research/pull/460: I needed the flexibility to read data from a CSV or get data from RabbitMQ and feed it message by message to the region.

cc @subutai @rhyolight 

Fixes: #3065

### CODE REVIEW

File: src/nupic/regions/RecordSensor.py [hunk 1]
```diff
@@ -556,3 +572,4 @@ def readFromProto(cls, proto):
     instance.numCategories = proto.numCategories
 
     return instance
+
```
Comment (scottpurdy) [inline src/nupic/regions/RecordSensor.py]: make sure files end with a newline. Many editors handle this automatically but YMMV

**Comment by scottpurdy** [comment]
:+1: after fixing the newline-at-end-of-file issue. Also, NuPIC issues need to have a "fixes #XXX" referencing an issue in order to pass the "Fixes Issue Validator"

**Comment by breznak** [comment]
ping? this is really just one line from finish :)

**Comment by scottpurdy** [comment]
@marionleborgne

**Comment by breznak** [comment]
// linked to the issue #3065

**Comment by rhyolight** [comment]
@marionleborgne What is the status of this? @scottpurdy has a simple change and then it can be merged?

**Comment by marionleborgne** [comment]
@rhyolight @scottpurdy @breznak sorry, this fell off my radar. I just added the new line. Thanks.

**Comment by scottpurdy** [comment]
:+1:

**Pull request merged**